session/send.go

```diff
@@ -19,13 +19,17 @@ package session
 import (
 	"fmt"
 	"io"
-	"sync"
 	"time"
 
 	cConstants "github.com/katzenpost/client/constants"
 	"github.com/katzenpost/core/constants"
 	"github.com/katzenpost/core/crypto/rand"
 	sConstants "github.com/katzenpost/core/sphinx/constants"
+	"gopkg.in/eapache/channels.v1"
+)
+
+const (
+	RoundTripTimeSlop time.Duration = 88 * time.Second
```

Comment:
Are you trying to go back to the future or did you benchmark this?
I think we should derive the "slop" factor from the difference between the mean fetch interval and mean message round trip time ( + some interval/std deviation from the mean to tune the rate of false drops).

Perhaps the client fetch interval should be derived from the mix parameters; message round trips could be then binned into a future fetch window, and considered dropped if they have not arrived by then. To rephrase: rather than setting specific timeouts per message, a schedule of fetch operations is generated prior to transmitting messages, and when transmitting the message, the message id is placed into a bucket of message id's associated with that message fetch operation. If the message response hasn't arrived by that fetch operation, it is considered dropped. Note that the message fetch frequency needs to match or exceed the message send frequency, and since we fetch one message per operation the particular fetch operation a message will be retrieved in cannot be predicted exactly.

Reply:
There is no mean fetch interval. Instead there is a precise fetch interval. Regular. Periodic. Not everything has to be a poisson process.